WordPress/gutenberg
# Navigator: focus screen wrapper instead of first tabbable element

## What?


This PR tweaks the way `Navigator` manages focus when navigating to a new screen.

## Why?


As discussed in #51762, the current focus restoration behavior of the `Navigator` component can cause some unexpected consequences depending on what component gets programmatically focused when navigating to a new screen.

For wider context: https://github.com/WordPress/gutenberg/pull/51762#issuecomment-1602360836

## How?


Prior to this PR, when navigating to a new screen, `Navigator` moves focus to either:
- the button that was clicked last in that screen, when navigating back
- otherwise, the first focusable item

The changes from this PR cause the "fallback" focus behavior to focus on the navigator screen wrapper, instead of the first focusable item.

## Testing Instructions





In the storybook's Navigator example:
- click on any button to navigate to another screen
- notice how the focus is moved to the screen wrapper. The wrapper is not visibly focused, but pressing the tab key will move focus on the first tabbable element inside the new screen
- navigate back to the previous screen. Notice how the focus is restored on the button that was previously clicked, as it already happens in trunk

In the side editor's sidebar:
- load the site editor
- navigate through the editor's screen
- notice how the "back" button is not automatically focused when a new screen is rendered. Pressing the tab key should move focus to the back button, though.

## Screenshots or screencast 

Trunk:

https://github.com/WordPress/gutenberg/assets/1083581/9066c2bd-d1a9-4d0d-a7e8-adf256cc0794

This PR:

https://github.com/WordPress/gutenberg/assets/1083581/2c10ad55-215e-4027-a770-c1ddb7a58d85
